social_engagement/handlers.py

```diff
@@ -11,107 +9,146 @@
     thread_deleted,
     comment_deleted,
     thread_voted,
-    comment_voted,
     thread_followed,
     thread_unfollowed,
+    thread_or_comment_flagged,
 )
 import lms.lib.comment_client as cc
-from social_engagement.tasks import task_update_user_engagement_score
+from social_engagement.tasks import task_handle_change_after_signal
 
 
 @receiver(thread_deleted)
-@receiver(comment_deleted)
 @receiver(thread_voted)
-@receiver(comment_voted)
-def thread_comment_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
+def thread_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
     """
-    Updates user social engagement score
+    Updates user social engagement score for deleted and voted thread (or voted comment).
+    TODO: And also, for some reason, on voted comment.
     """
-    post = kwargs['post']
-    course_id = getattr(post, 'course_id', None)
-    user_id = getattr(post, 'user_id', None)
-    social_engagement_enabled = settings.FEATURES.get('ENABLE_SOCIAL_ENGAGEMENT', False)
```

Comment:
Are we no longer doing this check up-front? If social engagement was disabled for whatever reason, we'd end up still getting a bunch of celery jobs queued up each time stats need to be updated, but which would finish themselves immediately because we now check for this in the job itself. I think we should just not schedule jobs if we don't have the feature enabled.